WalletWasabi.Tests/UnitTests/WabiSabi/Integration/WabiSabiHttpApiIntegrationTests.cs

```diff
@@ -180,6 +180,101 @@ public async Task SoloCoinJoinTestAsync(long[] amounts)
 		await roundStateUpdater.StopAsync(CancellationToken.None);
 	}
 
+	[Theory]
+	[InlineData(new long[] { 10_000_000, 20_000_000, 30_000_000, 40_000_000, 100_000_000 })]
+	public async Task SoloCoinJoinTaprootAsync(long[] amounts)
```

Comment:
I wouldn't use integration tests for almost anything. The behavior can be verified in a small UT after a tiny refactoring:

Create an extension method `GetNextDestinations` that receives the number of taproots and segwits and move the new code from `CoinJoinClient` to that extension method. Then test the new method.

Reply:
It is not really possible to test CoinJoinClient as a whole, only in the integration test. Extracting methods are breaking apart the CoinJoinClient and in the end, we could end up in the same way as we did now. The tests were only checking parts and the bug was in between. 
You know what? I can merge this taproot test into the existing SoloCoinJoinAsync - WDYT?

Reply:
Ok